docs/configdoc.md

```diff
@@ -0,0 +1,104 @@
+# L3AFD Config Options Documentation
+
+See [l3afd.cfg](https://github.com/l3af-project/l3af-arch/blob/main/dev_environment/cfg/l3afd.cfg) for a full example configuration.
+
+
+```
+[DEFAULT]
+
+[l3afd]
+pid-file: ./l3afd.pid
+datacenter: dummy
+bpf-dir: /dev/shm
+bpf-log-dir:
+kernel-major-version: 4
+kernel-minor-version: 15
+shutdown-timeout: 1s
+http-client-timeout: 10s
+max-nf-restart-count: 3
+bpf-chaining-enabled: true
+swagger-api-enabled: false
+# PROD | DEV
+environment: PROD
+....
+```
+
+### Below is the detailed documentation for each field
+
+
+## [l3afd]
+| FieldName     | Example       | Description     | Required        |
+| ------------- | ------------- | --------------- | --------------- |
+|pid-file| `"./l3afd.pid"`  | The pin Path for l3afd.pid file which contains process id of eBPF Programs | Yes |
+|datacenter| `"dummy"` | Name of Datacenter| Yes |
+|bpf-dir| `"/dev/shm"` | Absolute Path of eBPF Programs (where they are extracted) | Yes |
+|bpf-log-dir|`""`      | Absolute Path, where we store logs of eBPF Programs.We are not storing logs of l3afd actions on eBPF Programs| No |
+|kernel-major-version|`"4"`|Major version of the kernel| Yes(Only for linux) |
+|kernel-minor-version|`"15"`|Minor version of the kernel (Ex 4.15)| Yes(Only for linux) |
+|shutdown-timeout|`"1s"`|It is the amount of time allowed for L3afd graceful stop ( stopping l3afd).If L3afd graceful stop is not successful then os interrupt stop the l3afd| No |
```

Comment:
I find this description confusing as it sounds like this is configuring the OS.  Glancing at the code, it looks like this is configuring l3afd itself to exit after that time, rather than causing the OS to kill the l3afd process.

Reply:
I have changed.